### DISCUSSION TITLE
dot notation form control names not working

### DISCUSSION THREAD

**Discussion by keeganstothert** [react-hook-form/react-hook-form#2013] [Q&A] [answered]
[sandbox](https://codesandbox.io/embed/inspiring-sun-6w9yi?fontsize=14&hidenavigation=1&theme=dark)

as you can see in the above sandbox the dot notation is not working as i expected. when naming an input as `emails.68` I expect it to updated as `{ emails: { 68: "value" } }` instead it's nuking the default state and taking the dot literally as `{ "emails.68": "value" }`

**Comment by bluebill1049** [comment, marked as answer]
don't think we support numbers like that. To support that, `regex` would probably kill me.

**Reply by bluebill1049** [reply]
unless you happy to take a look the source and support this, i don't think it's trivial.

**Reply by keeganstothert** [reply]
Dang, everything is stored by it’s Id (numeric) in my redux store and I was hoping to map it one to one in my forms

**Reply by bluebill1049** [reply]
can you try this? `test[5]` not sure if this works on top of my head.

**Reply by keeganstothert** [reply]
that doesn't work either unfortunately

**Reply by bluebill1049** [reply]
guess you will have do a mapping on submit then..
